src/devices/nm-device.c

```diff
@@ -5421,13 +5422,41 @@ nm_device_match_parent_hwaddr (NMDevice *device,
 	return parent_mac && nm_utils_hwaddr_matches (setting_mac, -1, parent_mac, -1);
 }
 
+static gboolean
+glob_matches (const char *str, const char *const *patterns, guint length)
```

Comment:
maybe move this to `src/nm-core-utils.h`, because it's strongly related to `nm_match_spec_device()`. Possibly, we should use `fnmatch` instead of `g_pattern_match_simple()` (which would be an API change, but unlikely that anybody would notice).

I still wonder, whether we need to differenciate between no-match and negative-match. Clearly, your code does not. Will that also be correct when adding more matches? In other words, should the function should return `NMMatchSpecMatchType`?